client/js/chart/index.js

```diff
@@ -101,30 +87,62 @@ export function createChart(container, histories, dates) {
           }),
       );
 
-      activeSeries = hit;
-      activeX = x;
-      activeY = y;
+      hoverState.series = hit;
+      hoverState.x = x;
+      hoverState.y = y;
     }
   });
 
-  svg.addEventListener('mouseleave', () => {
-    if (activeSeries) {
-      const seriesElement = seriesElements[activeSeries];
+  const clearHover = () => {
+    if (hoverState.series !== null) {
+      const seriesElement = seriesElements[hoverState.series];
       seriesElement.dispatchEvent(
           new CustomEvent('series-clear', {bubbles: true}),
       );
     }
 
-    activeX = null;
-    activeY = null;
-  });
+    hoverState.series = null;
+    hoverState.x = null;
+    hoverState.y = null;
+  };
+
+  svg.addEventListener('mouseleave', () => clearHover());
+  scrollContainer.addEventListener('scroll', () => clearHover());
 
   const tooltip = createTooltip(container, svg, histories, dates);
 
   container.append(scrollContainer);
   container.append(tooltip);
 }
 
+/**
+ * Tests whether the mouse is currently near a point on the chart.
+ *
+ * @param {SVGSVGElement} svg The SVG element for the chart.
+ * @param {number[][]} rows The ranks of each series at each date.
+ * @param {number} clientX The x position of the mouse.
+ * @param {number} clientY The y position of the mouse.
+ * @returns {{hit: number | null, x: number | null, y:number | null}} A hit
+ * result.
+ */
+function hitTest(svg, rows, clientX, clientY) {
```

Comment:
You'll have a simpler time and likely more performant code if you just listen on `mousemove` of each run's parent `<g>` tag:
https://stackblitz.com/edit/typescript-brjdbu

Reply:
I believe this will pick up the invisible hit paths as well.

Reply:
That's what I did at first, but it resulted in behaviour that I didn't really want.

- If I just do `mousemove` events on the lines, then you have to land the cursor exactly on the line for anything to hit, which is very difficult.
- I had the idea of adding transparent thick lines on top of the visible thin lines, which I had in one of the previous PRs, but the problem with this is that wherever the lines overlap, then the behaviour will be inconsistent